ocean_lib/models/test/test_side_staking.py

```diff
@@ -405,3 +414,224 @@ def test_side_staking(
+    assert erc20_token.balanceOf(another_consumer_wallet.address) == 0
+    assert (
+        ocean_token.balanceOf(another_consumer_wallet.address)
+        <= initial_ocean_user_balance
+    )
+
+    # Test in a different order
+    initial_ocean_user_balance = ocean_token.balanceOf(another_consumer_wallet.address)
+    swap_exact_amount_in_base_token(
+        bpool, erc20_token, ocean_token, another_consumer_wallet, to_wei("20")
+    )
+    join_pool_one_side(web3, bpool, ocean_token, another_consumer_wallet)
+    swap_exact_amount_in_datatoken(
+        bpool,
+        erc20_token,
+        ocean_token,
+        another_consumer_wallet,
+        erc20_token.balanceOf(another_consumer_wallet.address),
+    )
+    wallet_exit_pool(web3, bpool, pool_token, another_consumer_wallet)
+    swap_exact_amount_in_datatoken(
+        bpool,
+        erc20_token,
+        ocean_token,
+        another_consumer_wallet,
+        erc20_token.balanceOf(another_consumer_wallet.address),
+    )
+
+    # Check that users hasn't made any profit
+    assert pool_token.balanceOf(another_consumer_wallet.address) == 0
+    assert erc20_token.balanceOf(another_consumer_wallet.address) == 0
+    assert (
+        ocean_token.balanceOf(another_consumer_wallet.address)
+        <= initial_ocean_user_balance
+    )
+
+
+def test_side_staking_constant_rate(
+    web3, config, publisher_wallet, consumer_wallet, another_consumer_wallet
+):
+    """
+    In this test we test that the side staking bot keeps the same rate when the datatoken supply is less than the cap.
+    """
+    # Test initial values and setups
+    initial_pool_liquidity = to_wei("1")
+    token_cap = to_wei("5")
+    swap_market_fee = to_wei("0.0001")
+    swap_fee = to_wei("0.0001")
+    big_allowance = to_wei("100000000")
+
+    ocean_token = ERC20Token(web3, get_address_of_type(config, "Ocean"))
+
+    ocean_token.transfer(another_consumer_wallet.address, to_wei("50"), consumer_wallet)
+
+    bpool, erc20_token, erc721_token, pool_token = create_nft_erc20_with_pool(
+        web3,
+        config,
+        publisher_wallet,
+        swap_fee,
+        swap_market_fee,
+        initial_pool_liquidity,
+        token_cap,
+    )
+
+    ocean_token.approve(bpool.address, big_allowance, another_consumer_wallet)
+    erc20_token.approve(bpool.address, big_allowance, another_consumer_wallet)
+
+    # End test initial values and setups
+
+    initial_spot_price_basetoken_datatoken = bpool.get_spot_price(
+        ocean_token.address, erc20_token.address, swap_market_fee
+    )
+    join_pool_one_side(web3, bpool, ocean_token, another_consumer_wallet, to_wei("0.5"))
+    final_spot_price_basetoken_datatoken = bpool.get_spot_price(
+        ocean_token.address, erc20_token.address, swap_market_fee
+    )
+
+    # TODO: Is this assertion normal?
+    assert (
+        final_spot_price_basetoken_datatoken != initial_spot_price_basetoken_datatoken
```

Comment:
The spot price is not identical, there is a 7% change.
Test case: 0 vesting amount, added base token 1 side liquidity when the liquidity was considerably below the cap